src/components/YAxis.tsx

```diff
@@ -0,0 +1,57 @@
+import { axisLeft } from 'd3-axis';
+import { select } from 'd3-selection';
+import React, { useLayoutEffect, useRef } from 'react';
+
+import { usePlotContext } from '../hooks';
+import type { AxisProps } from '../types';
+
+const YAxis = ({ label, fontSize = 16, showTicks }: AxisProps) => {
+  const axisRef = useRef(null);
+  const { yScale, margin, height, width } = usePlotContext();
+
+  useLayoutEffect(() => {
```

Comment:
I discussed with Daniel and we think we should follow React's official recommendations and `useEffect` instead.

Reply:
Just curious, why?

Reply:
https://reactjs.org/docs/hooks-reference.html#uselayouteffect